src/plugins/prepare.rs

```diff
@@ -490,3 +492,258 @@ fn clamp_collider_density(mut query: Query<&mut ColliderDensity, Changed<Collide
+                    (
+                        Position::from_xyz(-1., 3., -3.),
+                        Rotation(Quaternion::from_axis_angle(Vec3::X, 0.1)),
+                    )
+                }
+            };
+            let e_1_with_pos_and_rot = app.world.spawn((RigidBody::Dynamic, pos_1, rot_1)).id();
+
+            // Spawn an entity with only `Position`
+            let pos_2 = {
+                #[cfg(feature = "2d")]
+                {
+                    Position::from_xy(10., 1.)
+                }
+                #[cfg(feature = "3d")]
+                {
+                    Position::from_xyz(10., 1., 5.)
+                }
+            };
+            let e_2_with_pos = app.world.spawn((RigidBody::Dynamic, pos_2)).id();
+
+            // Spawn an entity with only `Rotation`
+            let rot_3 = {
+                #[cfg(feature = "2d")]
+                {
+                    Rotation::from_sin_cos(0.4, 0.5)
+                }
+                #[cfg(feature = "3d")]
+                {
+                    Rotation(Quaternion::from_axis_angle(Vec3::Z, 0.4))
+                }
+            };
+            let e_3_with_rot = app.world.spawn((RigidBody::Dynamic, rot_3)).id();
+
+            // Spawn entities with `Transform`
+            let trans_4 = {
+                Transform {
+                    translation: Vec3::new(-1.1, 6., -7.),
+                    rotation: Quat::from_axis_angle(Vec3::Y, 0.1),
+                    scale: Vec3::ONE,
+                }
+            };
+            let e_4_with_trans = app.world.spawn((RigidBody::Dynamic, trans_4)).id();
+
+            let trans_5 = {
+                Transform {
+                    translation: Vec3::new(8., -1., 0.),
+                    rotation: Quat::from_axis_angle(Vec3::Y, -0.1),
+                    scale: Vec3::ONE,
+                }
+            };
+            let e_5_with_trans = app.world.spawn((RigidBody::Dynamic, trans_5)).id();
+
+            // Spawn entity without any transforms
+            let e_6_without_trans = app.world.spawn(RigidBody::Dynamic).id();
+
+            // Spawn entity without a ridid body
+            let e_7_without_rb = app.world.spawn(()).id();
+
+            // Run the system
+            app.update();
+
+            // Check the results are as expected
+            if config.position_to_transform {
+                assert!(app.world.get::<Transform>(e_0_with_pos_and_rot).is_some());
+                let transform = app.world.get::<Transform>(e_0_with_pos_and_rot).unwrap();
+                let expected: Vec3 = {
+                    #[cfg(feature = "2d")]
+                    {
+                        pos_0.f32().extend(0.)
+                    }
+                    #[cfg(feature = "3d")]
+                    {
+                        pos_0.f32()
+                    }
+                };
+                assert_eq!(transform.translation, expected);
+                let expected = Quat::from(rot_0);
```

Comment:
Annoyingly, we need to do this currently
```suggestion
                let expected = Quaternion::from(rot_0).f32();
```

Reply:
5d94fba75cc343a4f791fe3aadbc65f38aa9b43a